transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -545,25 +545,25 @@ static void JNI_OnUnload_netty_transport_native_epoll0(JavaVM* vm, void* reserve
     netty_epoll_native_JNI_OnUnLoad(env);
 }
 
-// As we build with -fvisibility=hidden we need to ensure we mark the entry load and unload functions used by the
-// JVM as visible.
+// We build with -fvisibility=hidden so ensure we mark everything that needs to be visible with JNIEXPORT
+// http://mail.openjdk.java.net/pipermail/core-libs-dev/2013-February/014549.html
 
 // Invoked by the JVM when statically linked
-jint __attribute__((visibility("default"))) JNI_OnLoad_netty_transport_native_epoll(JavaVM* vm, void* reserved) {
+JNIEXPORT jint JNI_OnLoad_netty_transport_native_epoll(JavaVM* vm, void* reserved) {
```

Comment:
When I approved earlier I thought this was due to wanting the explicit behavior of JNIEXPORT on linux/solaris.  However it seems now that the windows behavior (which would obviously not work here, but bear with me)  is now:

```c
#define JNIEXPORT __declspec(dllexport)
#define JNIIMPORT __declspec(dllimport)
```

Since you'll be making this for tcnative, are you sure this is what you want?

Reply:
Yep that is exactly what I want 😉 Will verify what is exported on windows as well tho